circuits/cpp/src/aztec3/circuits/abis/types.hpp

```diff
@@ -17,4 +17,13 @@ template <typename NCT> struct PublicTypes {
     // used in schema serialization
     static constexpr char schema_name[] = "Public";  // NOLINT
 };
+
+template <typename B> inline void read(B& buf, verification_key& key)
+{
+    using serialize::read;
+    verification_key_data data;
+    read(buf, data);
+    key = verification_key{ std::move(data), barretenberg::srs::get_crs_factory()->get_verifier_crs() };
+}
+
```

Comment:
Why did this need to be added here?

Reply:
This code was originally present in circuits/cpp/src/aztec3/circuits/abis/previous_private_kernel_data.hpp and I removed it as this routine is present in Barretenberg.
It caused then issue in 'wasm cbinding', essentially any round trip from TS to C++ involving a data structure with a verification key were failing with a wasm error related to not finding this specific read method.
I added back it in this file back because this method is not specific at all to previous_private_kernel_data.hpp and because with the newly data structures and round-trip serialization tests I added, I got the same wasm error even if reverted back my original removal of this read method.
Having this method read in types.hpp solves this issue.